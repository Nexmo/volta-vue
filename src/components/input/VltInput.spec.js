import { shallowMount } from '@vue/test-utils';
import { VltInput } from '..';

describe('vlt-input', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(VltInput);
  });

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('is textarea', async () => {
    wrapper.setProps({ textarea: true });
    await wrapper.vm.$nextTick();
    expect(wrapper.element).toMatchSnapshot();
  });

  test('emits "test" on value change', () => {
    wrapper.find('input').setValue('test');
    expect(wrapper.emitted('input')).toEqual([['test']]);
  });
});
