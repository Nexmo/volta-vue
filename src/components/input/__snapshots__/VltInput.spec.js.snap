// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vlt-input is textarea 1`] = `
<div
  class="Vlt-input"
>
  <input
    type="text"
  />
   
  <!---->
   
  <!---->
</div>
`;

exports[`vlt-input renders correctly 1`] = `
<div
  class="Vlt-input"
>
  <input
    type="text"
  />
   
  <!---->
   
  <!---->
</div>
`;
